src/StreamJsonRpc.Tests/MessagePackFormatter.cs

```diff
@@ -47,13 +62,28 @@ public MessagePackFormatter(bool compress)
         }
 
         /// <inheritdoc/>
-        public JsonRpcMessage Deserialize(ReadOnlySequence<byte> contentBuffer)
+        public JsonRpc Rpc
         {
-            return this.compress
-                ? (JsonRpcMessage)LZ4MessagePackSerializer.Typeless.Deserialize(contentBuffer.AsStream())
-                : (JsonRpcMessage)MessagePackSerializer.Typeless.Deserialize(contentBuffer.AsStream());
+            private get => this.rpc;
+            set
+            {
+                Verify.Operation(this.rpc == null, "This formatter already belongs to another JsonRpc instance. Create a new instance of this formatter for each new JsonRpc instance.");
+
+                this.rpc = value;
+            }
         }
 
+        /// <summary>
+        /// Gets <see cref="MessagePackSerializerOptions.LZ4Default"/> if <see cref="compress"/> is true, otherwise <c>null</c>.
+        /// </summary>
+        private MessagePackSerializerOptions CompressOptions
+        {
+            get => this.compress ? MessagePackSerializerOptions.LZ4Default : null;
```

Comment:
>null [](start = 77, length = 4)

We must never return `null` here, since that means that the `MessagePack` library's (mutable) default will apply, but we must never trust a mutable default. We must be explicit. So this should be `MesasgePackSerializerOptions.Default` instead of `null`.